### PR TITLE
Add CBMC proof of updateBase64DecodingBuffer function

### PR DESCRIPTION
**Pull request by Bujain**
Add CBMC proof of updateBase64DecodingBuffer function

Description
-----------
Added CBMC proof files of updateBase64DecodingBuffer function

Checklist:
----------

- [x] I have tested my changes. No regression in existing tests.
- [x] My code is formatted using Uncrustify.
- [x] I have read and applied the rules stated in CONTRIBUTING.md. 

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by yanjos-dev — DISMISSED** [review]

File: test/cbmc/proofs/updateBase64DecodingBuffer/updateBase64DecodingBuffer_harness.c [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+    pBase64IndexBuffer = &base64IndexBuffer;
+    pNumDataInBuffer = &numDataInBuffer;
+
+    /* SYMBOL_TO_INDEX_MAP_VALUE_UPPER_BOUND is the maximum value of the
+     * base64Index which is enforced in ota_base64.c. */
+    __CPROVER_assume( base64Index < SYMBOL_TO_INDEX_MAP_VALUE_UPPER_BOUND );
+
+    /* The maximum number of data in the pBase64IndexBuffer should not
+     * exceed UINT32_MAX. */
+    __CPROVER_assume( numDataInBuffer < UINT32_MAX );
```
Comment (yanjos-dev) [inline test/cbmc/proofs/updateBase64DecodingBuffer/updateBase64DecodingBuffer_harness.c:35]: Is this check accomplishing anything? The data type for numDataInBuffer is a uint_32, so by definition it won't be able to go above the value of the UINT32_MAX right?

Comment (Bujain) [reply test/cbmc/proofs/updateBase64DecodingBuffer/updateBase64DecodingBuffer_harness.c:35]: If it takes the value UINT32_MAX, the increment on line(https://github.com/aws/ota-for-aws-iot-embedded-sdk/blob/2a8536e878139e8dde5a914581de88308ce2acda/source/ota_base64.c#L288) can cause integer overflow.

Comment (Bujain) [reply test/cbmc/proofs/updateBase64DecodingBuffer/updateBase64DecodingBuffer_harness.c:35]: In the ota code, the numDataInBuffer will not exceed `MAX_NUM_BASE64_DATA` which is 4.

Comment (yanjos-dev) [reply test/cbmc/proofs/updateBase64DecodingBuffer/updateBase64DecodingBuffer_harness.c:35]: I see, that makes sense. Thanks.

**Review by yanjos-dev — APPROVED** [review]

**Review by aggarg — APPROVED** [review]

**Pull request merged**